src/core-unit/Unit/Unit/Unit.Comparison.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace System;
+
+partial struct Unit
+{
+    public int CompareTo(Unit other) => 0;
+
+    public int CompareTo(object? obj) => obj switch
+    {
+        null => 1,
+        Unit => 0,
+        _ => throw new ArgumentException("The object is not the Unit.", nameof(obj))
+    };
+
+    public static int Compare(Unit left, Unit right) => (left, right) switch
```

Comment:
Will it be faster if we just return zero?

Reply:
Done: redundant switch is removed.